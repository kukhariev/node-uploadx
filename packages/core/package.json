{
  "name": "@uploadx/core",
  "version": "6.1.5",
  "description": "Node.js resumable upload middleware",
  "keywords": [
    "resumable",
    "upload",
    "uploadx",
    "express",
    "middleware"
  ],
  "author": {
    "name": "Oleg Kukhariev",
    "url": "https://github.com/kukhariev"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/kukhariev/node-uploadx.git"
  },
  "license": "MIT",
  "engines": {
    "node": ">=14.18.20"
  },
  "files": [
    "lib/*"
  ],
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "scripts": {
    "build": "tsc -b .",
    "clean": "rimraf ./tsconfig.tsbuildinfo ./lib"
  },
  "dependencies": {
    "bytes": "^3.1.0",
    "multiparty": "^4.2.2",
    "parse-duration": "^1.1.0"
  },
  "devDependencies": {
    "@types/bytes": "3.1.1",
    "@types/multiparty": "0.0.33"
  },
  "publishConfig": {
    "access": "public"
  }
}
