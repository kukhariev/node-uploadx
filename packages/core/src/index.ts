export * from './handlers';
export * from './storages';
export * from './utils';
export * from './types';
