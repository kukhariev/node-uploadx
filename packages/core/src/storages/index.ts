export * from './disk-storage';
export * from './file';
export * from './storage';
export * from './local-meta-storage';
export * from './meta-storage';
export * from './config';
