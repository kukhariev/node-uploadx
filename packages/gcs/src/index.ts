export * from './gcs-storage';
export * from './gcs-meta-storage';
export * from './constants';
