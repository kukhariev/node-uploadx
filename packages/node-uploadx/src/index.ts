export * from '@uploadx/core';
export * from '@uploadx/s3';
export * from '@uploadx/gcs';
