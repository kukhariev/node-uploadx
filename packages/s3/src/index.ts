export * from './s3-storage';
export * from './s3-meta-storage';
export * from './aws-error';
