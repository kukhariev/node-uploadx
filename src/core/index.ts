export * from './errors';
export * from './BaseHandler';
export * from './BaseStorage';
export * from './File';
export * from './Cors';
