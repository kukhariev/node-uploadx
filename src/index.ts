export * from './core';
export * from './core/utils';
export * from './Tus';
export * from './Uploadx';
export * from './Multipart';
