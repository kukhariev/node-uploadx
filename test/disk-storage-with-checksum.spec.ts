import { DiskStorageWithChecksum } from '../packages/core/src';
import { authRequest, metafile, storageOptions } from './shared';
import { Readable } from 'stream';

const directory = 'ds-test';

jest.mock('fs/promises');
jest.mock('fs');

describe('DiskStorageWithChecksum', () => {
  jest.useFakeTimers({ doNotFake: ['setTimeout'] }).setSystemTime(new Date('2022-02-02'));
  const options = { ...storageOptions, directory, checksum: 'sha1' as const };
  const req = authRequest();

  it('should support checksum resume from fs', async () => {
    let storage = new DiskStorageWithChecksum(options);
    let diskFile = await storage.create(req, { ...metafile });
    await storage.write({ ...diskFile, start: 0, body: Readable.from('01234') });
    storage = new DiskStorageWithChecksum(options);
    diskFile = await storage.create(req, { ...metafile });
    diskFile = await storage.write({ ...diskFile, start: 5, body: Readable.from('56789') });
    expect(diskFile).toMatchSnapshot();
  });
});
