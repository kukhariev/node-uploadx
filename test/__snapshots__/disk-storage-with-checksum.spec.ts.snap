// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiskStorageWithChecksum should support checksum resume from fs 1`] = `
Object {
  "bytesWritten": 10,
  "contentType": "video/mp4",
  "createdAt": "2022-02-02T00:00:00.000Z",
  "expiredAt": "2022-02-02T01:00:00.000Z",
  "id": "f7d13faa74e2475f-e8fed598250d10ea-7f59007b4b7cf67-120941ca7dc37b78",
  "metadata": Object {
    "custom": "",
    "lastModified": 1635398061454,
    "mimeType": "video/mp4",
    "name": "testfile.mp4",
    "sha1": "ZAPAntzKARqtb+j3B529GAOf3kI=",
    "size": 64,
  },
  "name": "userId/testfile.mp4",
  "originalName": "testfile.mp4",
  "sha1": "87acec17cd9dcd20a716cc2cf67417b71c8a7016",
  "size": 64,
  "status": "part",
  "userId": "userId",
}
`;
